src/node/transaction.cpp

```diff
@@ -36,10 +36,18 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
         if (!existingCoin.IsSpent()) return TransactionError::ALREADY_IN_CHAIN;
     }
     if (!node.mempool->exists(hashTx)) {
-        // Transaction is not already in the mempool. Submit it.
+        // Transaction is not already in the mempool.
         TxValidationState state;
-        if (!AcceptToMemoryPool(*node.mempool, state, std::move(tx),
-                nullptr /* plTxnReplaced */, false /* bypass_limits */, max_tx_fee)) {
+        CAmount fee{0};
+        if (AcceptToMemoryPool(*node.mempool, state, tx,
+                nullptr /* plTxnReplaced */, false /* bypass_limits */, /* test_accept */ true, &fee)) {
+            if (max_tx_fee && fee > max_tx_fee) {
+                return TransactionError::MAX_FEE_EXCEEDED;
+            }
+        }
```

Comment:
```suggestion
        if (max_tx_fee && AcceptToMemoryPool(*node.mempool, state, tx,
                nullptr /* plTxnReplaced */, false /* bypass_limits */, /* test_accept */ true, &fee)) {
            if (fee > max_tx_fee) {
                return TransactionError::MAX_FEE_EXCEEDED;
            }
        }
```
Performance: Call `AcceptToMemoryPool()` only if necessary.

Reply:
yep will do! :)